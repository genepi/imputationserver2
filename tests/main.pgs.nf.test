nextflow_pipeline {

    name "Pipeline Tests for PGS"
    script "main.nf"  

    test("Should run with pgs calculation") {

        when {
            params {
                project = "test-job"
                build = "hg19"
                files = "${projectDir}/tests/data/input/chr20-unphased/*.vcf.gz"
                allele_frequency_population = "eur"
                password = PASSWORD
                refpanel_yaml = "${projectDir}/tests/data/refpanels/hapmap2/cloudgene.yaml"
                output = "${outputDir}"

                pgs = [
                    enabled: true,
                    min_r2: 0,
                    fix_strand_flips: false                    
                ]

                encryption = [
                    enabled: false
                ]

                pgscatalog = [
                    scores: "${projectDir}/tests/data/input/pgs/test-scores.chr20.txt.gz",
                    meta: "${projectDir}/tests/data/input/pgs/test-scores.chr20.json"
                ]
            }
        }

        then {
            assert workflow.success

            //TODO: check which expected file is correct! (m3vcf vs msav)
            //def expected = csv("tests/data/input/chr20-unphased/scores.expected.txt")
            def expected = csv("tests/data/input/pgs/expected.txt")
            def actual = csv("${outputDir}/scores.txt")
            assertTableEquals actual, expected
        }

    }


    test("Should run with pgs calculation and category filter") {

        when {
            params {
                project = "test-job"
                build = "hg19"
                files = "${projectDir}/tests/data/input/chr20-unphased/*.vcf.gz"
                allele_frequency_population = "eur"
                password = PASSWORD
                refpanel_yaml = "${projectDir}/tests/data/refpanels/hapmap2/cloudgene.yaml"
                output = "${outputDir}"

                pgs = [
                    enabled: true,
                    min_r2: 0,
                    fix_strand_flips: false,
                    category: "Body measurement"              
                ]

                encryption = [
                    enabled: false
                ]

                pgscatalog = [
                    scores: "${projectDir}/tests/data/input/pgs/test-scores.chr20.txt.gz",
                    meta: "${projectDir}/tests/data/input/pgs/test-scores.chr20.json"
                ]
            }
        }

        then {
            assert workflow.success

            def log = file("${outputDir}/qc_report.txt")

            assert snapshot(
                log.text
            ).match()

            //TODO: check which expected file is correct! (m3vcf vs msav)
            //def expected = csv("tests/data/input/chr20-unphased/scores.expected.txt")
            def expected = csv("tests/data/input/pgs/expected.txt")
            
            def actual = csv("${outputDir}/scores.txt")
            with(actual){
                assert columnNames == ["sample", "PGS000027"]
                assert columns["sample"] == expected.columns["sample"]
                assertArrayEquals columns["PGS000027"], expected.columns["PGS000027"]
            } 

        }

    }

}
