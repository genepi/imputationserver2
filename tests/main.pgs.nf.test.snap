{
    "Should run with pgs calculation and category filter": {
        "content": [
            "[{\"command\":\"LOG\",\"params\":[\"Reference Panel Ranges: genome-wide\"]},{\"command\":\"BEGIN_TASK\",\"params\":[\"Running Calculating QC Statistics...\"]},{\"command\":\"UPDATE_TASK\",\"params\":[\"Calculating QC Statistics [1/1]\\n\\nAnalyze file chr20.R50.merged.1.330k.recode.unphased.small.vcf.gz...\",3]},{\"command\":\"END_TASK\",\"params\":[\"Calculating QC Statistics\",0]},{\"command\":\"MESSAGE\",\"params\":[\"\\u003cb\\u003eStatistics:\\u003c/b\\u003e \\nAlternative allele frequency \\u003e 0.5 sites: 2,296\\nReference Overlap: 99.00 %\\nMatch: 7,735\\nAllele switch: 0\\nStrand flip: 0\\nStrand flip and allele switch: 0\\nA/T, C/G genotypes: 0\\n\\u003cb\\u003eFiltered sites:\\u003c/b\\u003e \\nFilter flag set: 0\\nInvalid alleles: 0\\nMultiallelic sites: 0\\nDuplicated sites: 0\\nNonSNP sites: 0\\nMonomorphic sites: 11\\nAllele mismatch: 0\\nSNPs call rate \\u003c 90%: 0\",0]},{\"command\":\"MESSAGE\",\"params\":[\"Excluded sites in total: 11\\nRemaining sites in total: 7,735\\nSee snps-excluded.txt for details\\nTyped only sites: 78\\nSee typed-only.txt for details\\n\",2]}]"
        ],
        "meta": {
            "nf-test": "0.9.0-rc2",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-28T10:58:28.192235"
    }
}